### PR TITLE
Add FFU support

### PR DESCRIPTION
**Pull request by arthur-simchaev-sndk**
Add FFU (Field Firmware Update) support accroding to section
11.3.28.2.1 of UFS 2.1 spec

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # SPDX-License-Identifier: GPL-2.0-or-later
 # Copyright (C) 2019 Western Digital Corporation or its affiliates
 
-export CC := $(CROSS_COMPILE)gcc
-
+#export CC := $(CROSS_COMPILE)gcc
+CC := /usr/local/sandisk/toolchain/gcc-linaro-aarch64-linux-gnu-4.9-2014.09_linux/bin/aarch64-linux-gnu-gcc
```
Comment (beanhuo) [inline Makefile]: hi, Arthur
I think you forgot to delete this your own toolchain location specification.

Comment (arthur-simchaev-sndk) [reply Makefile]: Thank you for your comment. Fixed it.

Comment (AviShchislowski-zz) [reply Makefile]: remove hard coded CC path

**Review by AviShchislowski-zz — APPROVED** [review]

File: ufs_ffu.c [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+};
+
+
+extern int do_query_rq(int fd, struct ufs_bsg_request *bsg_req,
+		struct ufs_bsg_reply *bsg_rsp, __u8 query_req_func,
+		__u8 opcode, __u8 idn, __u8 index, __u8 sel,
+		__u16 req_buf_len, __u16 res_buf_len, __u8 *data_buf);
+extern struct desc_field_offset device_desc_field_name[];
+
+/* Get sense key string or NULL if not available */
```
Comment (avri-altman-sndk) [inline ufs_ffu.c:42]: this can be more economically done using a static array, or even better stringify the enum

Comment (avri-altman-sndk) [reply ufs_ffu.c:42]: but this is really just a nitpic, the rest of the code looks fine, so better we'll fix it in some other time, in a different patch

**Review by avri-altman-sndk — APPROVED** [review]

**Pull request merged**